sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/StateHolder.java

```diff
@@ -95,11 +105,43 @@ static boolean getLoadStateFeatureFlag(String name) {
     static void setLoadState(String name, Boolean loaded) {
         LOAD_STATE.put(name, loaded);
     }
-    
+
     /**
      * @param LOAD_STATE the loadState to set
      */
     static void setLoadStateFeatureFlag(String name, Boolean loaded) {
         setLoadState(name + FEATURE_ENDPOINT, loaded);
     }
+
+    /**
+     * @return the nextForcedRefresh
+     */
+    public static Instant getNextForcedRefresh() {
+        return nextForcedRefresh;
+    }
+
+    /**
+     * Set after load or refresh is successful.
+     * @param nextForcedRefresh the nextForcedRefresh to set
+     */
+    public static void setNextForcedRefresh(Duration refreshPeriod) {
+        nextForcedRefresh = Instant.now().plusSeconds(refreshPeriod.getSeconds());
+    }
+
+    /**
+     * Sets a minimum value until the next refresh. If a refresh interval has passed or is smaller than the calculated
+     * backoff time, the refresh interval is set to the backoff time.
+     * @param refreshInterval period between refreshe checks.
+     * @param properties Provider properties for min and max backoff periods.
+     */
+    static void resetAll(Duration refreshInterval, AppConfigurationProviderProperties properties) {
+        nextForcedRefresh = CalculatedBackoffTime.calculateBefore("client", nextForcedRefresh, refreshInterval, properties);
```

Comment:
Why do we use hardcoded "client" as endpoint here?

Reply:
It is a constant for retry attempts for the client refresh interval vs the endpoint refresh interval. I can make it a constant.

Reply:
Remind me again what client refresh interval means? Each endpoint can have a refresh interval (so one refresh interval per AppConfig client), and we now added a new overall refresh interval which would reload config from each endpoint. Is the latter what you are calling "client refresh interval"?

Reply:
This should be more clear now with the move to State holding the refresh attempts.